src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AadPlugin.cs

```diff
@@ -360,38 +360,45 @@ public Task<WebTokenRequest> CreateWebTokenRequestAsync(WebAccountProvider provi
             return msalTokenResponse;
         }
 
-        public string MapTokenRequestError(WebTokenRequestStatus status, uint errorCode, bool isInteractive)
+        public Tuple<string, bool> MapTokenRequestError(WebTokenRequestStatus status, uint errorCode, bool isInteractive)
         {
             if (status == WebTokenRequestStatus.UserInteractionRequired)
             {
-                return MsalError.InteractionRequired;
+                return Tuple.Create(MsalError.InteractionRequired, false);
             }
 
             if (status == WebTokenRequestStatus.ProviderError)
             {
                 if (errorCode == 0xcaa20005)
-                    return "WAM_server_temporarily_unavailable";
+                    return Tuple.Create("WAM_server_temporarily_unavailable", true);
 
                 unchecked // as per https://stackoverflow.com/questions/34198173/conversion-of-hresult-between-c-and-c-sharp
                 {
                     var hresultFacility = (((errorCode) >> 16) & 0x1fff);
-                    if (hresultFacility == 0xAA3 // FACILITY_ADAL_HTTP in AAD WAM plugin
-                         || hresultFacility == 0xAA7 // FACILITY_ADAL_URLMON in AAD WAM plugin
-                         || hresultFacility == 0xAA8) // FACILITY_ADAL_INTERNET in AAD WAM plugin
+                    switch (hresultFacility)
                     {
-                        return "WAM_no_network";
-                    }
+                        case 0xAA3: // FACILITY_ADAL_HTTP in AAD WAM plugin
+                        case 0xAA7: // FACILITY_ADAL_URLMON in AAD WAM plugin
+                        case 0xAA8: // FACILITY_ADAL_INTERNET in AAD WAM plugin
```

Comment:
I think you got the hrresultFacility and the error code mixed up. They are different numbers. 

There are 4 facilities which are retry-able - 
FACILITY_ADAL_HTTP 
FACILITY_ADAL_URLMON 
FACILITY_ADAL_INTERNET  
FACILITY_ADAL_BACKGROUND_INFRASTRUCTURE

And a bunch of **error codes** which are retry-able

```csharp
 return (errorCode == BT_E_SPURIOUS_ACTIVATION // background infrastructure canceled, due not enough resources at this moment. 

            || errorCode == WAIT_TIMEOUT // overall request timed out. 

            || errorCode == HRESULT_FROM_WIN32(WAIT_TIMEOUT) // overall request timed out. 

            || errorCode == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND) // bug in WAM core in some version of Windows, the issues is retriable Bug 15107486. 

            || errorCode == ERROR_ADAL_SERVER_ERROR_TEMPORARILY_UNAVAILABLE // server temporary unavailable. 

            || errorCode == ERROR_ADAL_SERVER_ERROR_RECEIVED // internal server error. 

```

Reply:
Yes got those mixed up. Changed the logic to keep them separate.

Reply:
Will add the constants